### PR TITLE
RUBY-2007 Add AWS KMS options to Data Key Context

### PR DESCRIPTION
**Pull request by egiurleo**
The second in a series of a few pull requests.

### CODE REVIEW

File: lib/mongo/crypt/binding.rb [hunk 1]
```diff
@@ -423,6 +423,72 @@ def self.ctx_setopt_algorithm(context, name)
         end
       end
 
+      # Configure the ctx to take a masterkey from AWS
```
Comment (p-mongo) [inline lib/mongo/crypt/binding.rb]: "master key" please

File: lib/mongo/crypt/data_key_context.rb [hunk 2]
```diff
@@ -25,18 +25,104 @@ class DataKeyContext < Context
...
+            "Valid options are 'aws' and 'local'"
+          )
+        end
+
+        super(mongocrypt, io)
+
+        Binding.ctx_setopt_masterkey_local(self) if kms_provider == 'local'
+
+        if kms_provider == 'aws'
+          masterkey_opts = options[:masterkey]
```
Comment (p-mongo) [inline lib/mongo/crypt/data_key_context.rb]: `master_key` please in both

File: lib/mongo/crypt/data_key_context.rb [hunk 3]
```diff
@@ -25,18 +25,104 @@ class DataKeyContext < Context
...
+        unless ['aws', 'local'].include?(kms_provider)
+          raise ArgumentError.new(
+            "#{kms_provider} is an invalid kms provider. " +
+            "Valid options are 'aws' and 'local'"
+          )
+        end
+
+        super(mongocrypt, io)
+
+        Binding.ctx_setopt_masterkey_local(self) if kms_provider == 'local'
```
Comment (p-mongo) [inline lib/mongo/crypt/data_key_context.rb]: Can this and the following if statement be organized into a single statement, either case/when or if/elsif please.

File: lib/mongo/crypt/data_key_context.rb (hunk 2 above)
Comment (egiurleo) [reply lib/mongo/crypt/data_key_context.rb]: 👍 Will audit all my future pull requests for this.

**Pull request merged**